### PR TITLE
Add `graph create` command to create named subgraphs

### LINKED ISSUE #181: File paths in output are broken (e.g. `foo/<src>bar`)
See the `<src>` in the following output:
```
➜  decentraland git:(jannis/update-to-save-getById) ✗ yarn build --debug
yarn run v1.12.3
warning package.json: No license field
$ graph build subgraph.yaml --debug

1/3 Load subgraph: subgraph.yaml

2/3 Compile subgraph
Compile data source mapping: LANDRegistry => /Users/jannis/Work/graphprotocol/subgraphs/decentraland/dist/LANDRegistry/LANDRegistry<src>wasm
Compile data source mapping: Marketplace => /Users/jannis/Work/graphprotocol/subgraphs/decentraland/dist/Marketplace/Marketplace<src>wasm

3/3 Write compiled subgraph to output directory
Copy subgraph file: /Users/jannis/Work/graphprotocol/subgraphs/decentraland/dist/schema<src>graphql
Write subgraph file: /Users/jannis/Work/graphprotocol/subgraphs/decentraland/dist/LANDRegistry/abis/LANDRegistry<src>json
Write subgraph file: /Users/jannis/Work/graphprotocol/subgraphs/decentraland/dist/Marketplace/abis/Marketplace<src>json
Write subgraph manifest: /Users/jannis/Work/graphprotocol/subgraphs/decentraland/dist/subgraph<src>yaml

Build completed

Subgraph: /Users/jannis/Work/graphprotocol/subgraphs/decentraland/dist/subgraph.yaml
```

### LINKED ISSUE #189: Add `graph create` command
This should add a `graph create` command that takes a subgraph name and calls a graph node's `subgraph_create` JSON-RPC method to register the name.

### LINKED ISSUE #191: graph remove has old flag `--api-key`
should be `--access-token`

### LINKED ISSUE #192: Change `graph deploy` subgraph name from flag to positional argument
Make deploy syntax more consistent with create/remove

### PR DESCRIPTION
**Pull request by Jannis**
Resolves #189. Fixes #191. Fixes #181. Resolves #192.

This also changes the `graph remove` and `graph deploy` command. Together, all three commands now have the following signature:

```sh
graph create <SUBGRAPH_NAME> ... # -g/--node, --access-token are supported
graph remove <SUBGRAPH_NAME> ... # -g/--node, --access-token are supported
graph deploy <SUBGRAPH_NAME> ... # -g/--node, --access-token are supported
```

### CODE REVIEW

**Review by timmclean — CHANGES_REQUESTED** [review]
It feels like `graph deploy` should be `graph deploy [SUBGRAPH_NAME] [MANIFEST]` now for consistency

File: graph.js [hunk 1]
```diff
@@ -320,16 +338,60 @@ app
   })
 
 app
-  .command('remove')
+  .command('create [SUBGRAPH_NAME]')
+  .description('Creates a named subgraph name')
+  .option('-g, --node <URL>[:PORT]', 'Graph node to create the subgraph in')
```
[the remark this answers is not in the document]
Comment (Jannis) [reply graph.js]: That is... true. I'll update.

**Comment by Jannis** [comment]
@timmclean I do agree about `graph deploy`. Can make that change in a separate PR though. Would you mind creating an issue for that?

**Comment by timmclean** [comment]
Done! https://github.com/graphprotocol/graph-cli/issues/192

File: graph.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jannis) [reply graph.js]: Done

**Comment by Jannis** [comment]
@timmclean Ready for another review pass. I've also added a bunch of other small improvements and fixes.

File: graph.js [hunk 2]
```diff
@@ -320,16 +323,60 @@ app
...
-  .command('remove')
+  .command('create [SUBGRAPH_NAME]')
+  .description('Creates a named subgraph name')
+  .option('-g, --node <URL>', 'Graph node to create the subgraph in')
+  .option('--access-token <TOKEN>', 'Graph access token')
+  .action(async (subgraphName, cmd) => {
+    if (subgraphName === undefined || cmd.node === undefined) {
+      console.error('Cannot create the subgraph')
+      console.error('--')
+      outputNameAndNodeConfig(cmd, { subgraphNameFromFlag: false, subgraphName })
```
Comment (timmclean) [inline graph.js]: should `subgraphNameFromFlag` still be here?

File: graph.js [hunk 3]
```diff
@@ -320,16 +323,60 @@ app
...
-  .action(cmd => {
-    if (cmd.subgraphName === undefined || cmd.node === undefined) {
+  .option('-g, --node <URL>', 'Graph node to remove the subgraph from')
+  .option('--access-token <TOKEN>', 'Graph access token')
+  .action(async (subgraphName, cmd) => {
+    if (subgraphName === undefined || cmd.node === undefined) {
       console.error('Cannot remove the subgraph')
       console.error('--')
-      outputNameAndNodeConfig(cmd)
+      outputNameAndNodeConfig(cmd, { subgraphNameFromFlag: false, subgraphName })
```
Comment (timmclean) [inline graph.js]: ditto on `subgraphNameFromFlag`

**Comment by timmclean** [comment]
@Jannis re-reviewed

**Comment by Jannis** [comment]
@timmclean Good catch. Removed the flag from all calls now.

**Review by timmclean — APPROVED** [review]

**Pull request merged**